libs/core/device-initialization/actions/initialize-mudita-pure.action.ts

```diff
@@ -39,20 +39,44 @@ export const initializeMuditaPure = createAsyncThunk<
     dispatch(
       setDeviceInitializationStatus(DeviceInitializationStatus.Initializing)
     )
+    if (!isActiveDeviceAttachedSelector(getState())) {
+      dispatch(
+        setDeviceInitializationStatus(DeviceInitializationStatus.Aborted)
+      )
+      return DeviceInitializationStatus.Aborted
+    }
+
     await dispatch(loadSettings())
```

Comment:
got an idea for it :D 

- in every async action necessary to be checked add this check if !isActiveDeviceAttachedSelector(getState()) and if this true then return rejectWithValue
- in this action 'initializeMuditaPure' check result if inner action returned reject then also reject it
- in place where initializeMuditaPure is invoked check if you got rejected and if yes then execute code below:
`dispatch(
        setDeviceInitializationStatus(DeviceInitializationStatus.Aborted)
      )
      return DeviceInitializationStatus.Aborted`
- if there are few places of invoking initializeMuditaPure then wrap it into function/thunk
      
 In my honest opinion, checking it everytime is overkill, I would suggest to simply reject in every async thunk when something is wrong and then act only when there is a problem. You are not counteracting problems, you simply break execution of this main thunk. So it will be the same.